### PR TITLE
Update native AOT static library guidance for .NET 8

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
## Summary

Reflects changes in https://github.com/dotnet/runtime/pull/89291.

Cc @dotnet/ilc-contrib

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Comment by BillWagner** [comment]
Hi @MichalStrehovsky 

Once you address the CI warnings, this is ready to merge.

**Comment by MichalStrehovsky** [comment]
> Hi @MichalStrehovsky
> 
> Once you address the CI warnings, this is ready to merge.

Is it possible to suppress the "inline html" warning. I need something to declutter the page. Or is there and equivalent of per-.net-version tabs?

File: core/nativeaot/NativeLibrary/README.md [hunk 1]
```diff
@@ -97,11 +97,14 @@ where `<Configuration>` is your project configuration (such as Debug or Release)
...
 The above command will drop a static library (Windows `.lib`, OSX/Linux `.a`) in `./bin/[configuration]/netstandard2.0/[RID]/publish/` folder and will have the same name as the folder in which your source file is present.
 
+<details>
+<summary>Extra requirements to link static libraries produced by .NET 7 (click to expand)</summary>
 When linking the generated static library, it is important to also include additional framework dependencies in the linker settings, and add `NativeAOT_StaticInitialization` to the symbol table. This can be accomplished by appending the following flag to the linker settings:
 
 * Windows: `/INCLUDE:NativeAOT_StaticInitialization`
 * Linux: `-Wl,--require-defined,NativeAOT_StaticInitialization`
 * macOS: `-Wl,-u,_NativeAOT_StaticInitialization`
+</details>
```
Comment (BillWagner) [inline core/nativeaot/NativeLibrary/README.md:107]: Add blank line for linter, and re-enable the warning:

```suggestion

</details>
<!-- markdownlint-enable MD033 -->
```

File: core/nativeaot/NativeLibrary/README.md [hunk 2]
```diff
@@ -97,11 +97,14 @@ where `<Configuration>` is your project configuration (such as Debug or Release)
 
 The above command will drop a static library (Windows `.lib`, OSX/Linux `.a`) in `./bin/[configuration]/netstandard2.0/[RID]/publish/` folder and will have the same name as the folder in which your source file is present.
 
+<details>
```
Comment (BillWagner) [inline core/nativeaot/NativeLibrary/README.md:100]: Here's how to disable the markdown lint warning:

```suggestion
<!-- markdownlint-disable MD033 -->
<details>
```

**Review by BillWagner — COMMENTED** [review]
Hi @MichalStrehovsky 

In this instance, let's disable the lint warning. There are zone pivots on the learn platform, but the sample readme files aren't published in that mode. 

I added suggestions to disable and enable the warnings. 

Once those are tested, I'll merge it.

**Comment by MichalStrehovsky** [comment]
> I added suggestions to disable and enable the warnings.

Thank you! Looks like they did the trick!

**Pull request merged**